src/transformers/models/vitpose/image_processing_vitpose.py

```diff
@@ -0,0 +1,566 @@
+        ]
+
+        data = {"pixel_values": images}
+        encoded_inputs = BatchFeature(data=data, tensor_type=return_tensors)
+
+        return encoded_inputs
+
+    def keypoints_from_heatmaps(
+        self,
+        heatmaps,
+        center,
+        scale,
+        kernel=11,
+        use_udp=False,
+    ):
+        """
+        Get final keypoint predictions from heatmaps and transform them back to
+        the image.
+
+        Args:
+            heatmaps (`np.ndarray` of shape `(batch_size, num_keypoints, height, width])`):
+                Model predicted heatmaps.
+            center (np.ndarray[N, 2]):
+                Center of the bounding box (x, y).
+            scale (np.ndarray[N, 2]):
+                Scale of the bounding box wrt height/width.
+            post_process (str/None):
+                Choice of methods to post-process heatmaps.
+                Currently supported: None, 'default', 'unbiased', 'megvii'.
+            unbiased (bool)
+                Option to use unbiased decoding. Mutually exclusive with megvii.
+                Note: this arg is deprecated and unbiased=True can be replaced
+                by post_process='unbiased'. Paper ref: Zhang et al. Distribution-Aware Coordinate
+                Representation for Human Pose Estimation (CVPR 2020).
+            kernel (int):
+                aussian kernel size (K) for modulation, which should match the heatmap gaussian sigma when training.
+                K=17 for sigma=3 and k=11 for sigma=2.
+            valid_radius_factor (float):
+                The radius factor of the positive area in classification heatmap for UDP.
+            use_udp (bool):
+                Use unbiased data processing.
+            target_type (str):
+                'GaussianHeatmap' or 'CombinedTarget'.
+                GaussianHeatmap: Classification target with gaussian distribution.
+                CombinedTarget: The combination of classification target
+                (response map) and regression target (offset map).
+                Paper ref: Huang et al. The Devil is in the Details: Delving into
+                Unbiased Data Processing for Human Pose Estimation (CVPR 2020).
+
+        Returns:
+            tuple: A tuple containing keypoint predictions and scores.
+
+            - preds (np.ndarray[batch_size, num_keypoints, 2]):
+                Predicted keypoint location in images.
+            - maxvals (np.ndarray[batch_size, num_keypoints, 1]):
+                Scores (confidence) of the keypoints.
+        """
+        # Avoid being affected
+        heatmaps = heatmaps.numpy().copy()
+
+        batch_size, num_keypoints, height, width = heatmaps.shape
+
+        preds, maxvals = _get_max_preds(heatmaps)
+
+        preds = post_dark_udp(preds, heatmaps, kernel=kernel)
+
+        # Transform back to the image
+        for i in range(batch_size):
+            preds[i] = transform_preds(preds[i], center[i], scale[i], [width, height], use_udp=use_udp)
```

Comment:
All image dimensions passed for transforms should be passed in (height, width) order